sdk/keyvault/keyvault-admin/src/lro/restore/operation.ts

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortSignalLike } from "@azure/abort-controller";
+import { PollOperationState, PollOperation } from "@azure/core-lro";
+import { OperationOptions, RequestOptionsBase } from "@azure/core-http";
+import { KeyVaultClient } from "../../generated/keyVaultClient";
+import {
+  KeyVaultClientFullBackupStatusResponse,
+  KeyVaultClientFullRestoreOperationOptionalParams
+} from "../../generated/models";
+import { createSpan, setParentSpan } from "../../tracing";
+import { KeyVaultClientFullRestoreOperationResponse } from "../../generated/models";
+
+/**
+ * An interface representing the publicly available properties of the state of a restore Key Vault's poll operation.
+ */
+export interface RestoreOperationState extends PollOperationState<undefined> {
+  /**
+   * The base URL to the vault
+   */
+  vaultUrl: string;
+  /**
+   * The URI of the blob storage account.
+   */
+  blobStorageUri: string;
+  /**
+   * The SAS token.
+   */
+  sasToken: string;
+  /**
+   * The Folder name of the blob where the previous successful full backup was stored.
+   */
+  folderName: string;
+  /**
+   * Identifier for the full restore operation.
+   */
+  jobId?: string;
+  /**
+   * Status of the restore operation.
+   */
+  status?: string;
+  /**
+   * The status details of restore operation.
+   */
+  statusDetails?: string;
+  /**
+   * The start time of the restore operation in UTC
+   */
+  startTime?: Date;
+  /**
+   * The end time of the restore operation in UTC
+   */
+  endTime?: Date;
+}
+
+/**
+ * An internal interface representing the state of a restore Key Vault's poll operation.
+ * @internal
+ */
+export interface RestorePollOperationState extends PollOperationState<undefined> {
+  /**
+   * Options for the core-http requests.
+   */
+  requestOptions?: RequestOptionsBase;
+  /**
+   * An interface representing the internal KeyVaultClient.
+   */
+  client?: KeyVaultClient;
+  /**
+   * The base URL to the vault
+   */
+  vaultUrl: string;
+  /**
+   * The URI of the blob storage account.
+   */
+  blobStorageUri: string;
+  /**
+   * The SAS token.
+   */
+  sasToken: string;
+  /**
+   * The Folder name of the blob where the previous successful full backup was stored
+   */
+  folderName: string;
+  /**
+   * The id returned as part of the restore request
+   */
+  jobId?: string;
+  /**
+   * Status of the restore operation.
+   */
+  status?: string;
```

Comment:
doesn't this always have some kind of known status, even if it's like "started" or something?

Reply:
Also it would be nice if this was a string union, but maybe we don't have that kind of safety?

Reply:
I only set this to started after I make the first request 🤔

Reply:
This property comes from the service! And, the swagger doesn't have a set of possible values :/ I could ask though. What do you think after this information?

Reply:
Note: this property is not part of core-lro, it's part of the service we're using for this client.